api/s2n.h

```diff
@@ -1269,6 +1269,24 @@ S2N_API extern int s2n_config_set_ticket_decrypt_key_lifetime(struct s2n_config
  */
 S2N_API extern int s2n_config_add_ticket_crypto_key(struct s2n_config *config, const uint8_t *name, uint32_t name_len,
         uint8_t *key, uint32_t key_len, uint64_t intro_time_in_seconds_from_epoch);
+
+/**
+ * Toggles ticketed resumption for TLS1.2 connections. Clients should not expect
+ * session tickets from servers and servers will not send session tickets when TLS1.2
+ * is negotiated and legacy tickets are disabled. Tickets will continue to be produced
+ * when TLS1.3 is negotiated and legacy tickets are disabled.
+ * 
+ * @note Disabling legacy tickets is a security improvement as TLS1.2 tickets
+ * are not forward secret.
+ * 
+ * @note The default behavior is that legacy tickets are enabled.
+ *
+ * @param config The config object being updated
+ * @param enabled Indicates if legacy tickets are enabled or disabled
+ * @returns S2N_SUCCESS on success. S2N_FAILURE on failure
+ */
+S2N_API extern int s2n_config_legacy_tickets(struct s2n_config *config, bool enabled);
+
```

Comment:
Do we want to make this more flexible and set a minimum version for tickets instead?

Reply:
Another option would be to be more explicit about the goals of this API, which is to ensure forward secrecy of sessions. So maybe something like `s2n_config_enforce_ticket_forward_secrecy(s2n_config, bool)`?

Reply:
But wouldn't that mean we'd still need another separate API if TLS1.4 fixes some currently unknown issue with TLS1.3 session resumption?

Reply:
I like `...enforce_ticket_forward_secrecy`. To me `s2n_config_legacy_tickets` sounds like a ticket version thing, "use an old version of the tickets". It also prevents us from needing another API if we ever support TLS 1.3 session resumption without DHE.

> But wouldn't that mean we'd still need another separate API if TLS1.4 fixes some currently unknown issue with TLS1.3 session resumption?

I think we'd need a new API in that case anyways? Because if some customer
1. wants to only use TLS 1.2 and TLS 1.3
2. forbids TLS 1.2 resumption (calls `s2n_config_legacy_tickets(false)`)
and then we changed the behavior of this API to also forbid TLS 1.3 stuff, wouldn't that be a breaking change?

Reply:
Okay it seems like the conclusion of this is the API should be s2n_config_enforce_ticket_forward_secrecy(bool)